modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -15,21 +15,36 @@ const fetchTextData = async (url: string, errMsg: string) => {
   return res.text();
 };
 
-const getOASFileUrl = (gitHash: string) => `${OAS_FILE_SERVER}${gitHash}.json`;
+interface AtlasSpecUrlParams {
+  apiKeyword: string;
+  apiVersion?: string;
+  resourceVersion?: string;
+}
 
-const getAtlasSpecUrl = async (apiKeyword: string) => {
+const getAtlasSpecUrl = async ({ apiKeyword, apiVersion, resourceVersion }: AtlasSpecUrlParams) => {
   // Currently, the only expected API fetched programmatically is the Cloud Admin API,
   // but it's possible to have more in the future with varying processes.
   const keywords = ['cloud'];
   if (!keywords.includes(apiKeyword)) {
     throw new Error(`${apiKeyword} is not a supported API for building.`);
   }
 
+  let versionExtension = '';
+
+  if (apiVersion) {
```

Comment:
const versionExtension = apiVersion && resourceVersion ? versionExtension += `-${resourceVersion}` : versionExtension += `-v${apiVersion.split('.')[0]}`

Reply:
Hey Yusuf! Thank you for the comment, I made a slight modification to what you provided, but I like the idea 👍